lib/twitter/accounts.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Twitter.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+  alias Twitter.Repo
+  alias Twitter.Accounts.User
+
+  @doc """
+  create user: validate parameters and insert in the db
+  """
+  @spec create_user(map()) ::
+          {:ok, User.t()} | {:error, Ecto.Changeset.t(User.t())}
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.registration_changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  get_user: returns user by id
+  """
+  @spec get_user(integer() | String.t()) :: User.t() | nil
+  def get_user(id) do
+    Repo.get(User, id)
+  end
+
+  @doc """
+  returns user by id or raises Ecto.NoResultsError if not found
+  """
+  @spec get_user!(integer() | String.t()) :: User.t()
+  # has error handling
+  def get_user!(id) do
```

Comment:
this public function should also have spec.

Reply:
sorry, not sure if I understand this comment.

Reply:
please disregard this comment. Did not see this spec 🙇